src/moscot/problems/space/_mapping.py

```diff
@@ -299,24 +319,51 @@ def solve(
         - :attr:`solutions` - the :term:`OT` solutions for each subproblem.
         - :attr:`stage` - set to ``'solved'``.
         """
-        return super().solve(  # type: ignore[return-value]
-            alpha=alpha,
-            epsilon=epsilon,
-            tau_a=tau_a,
-            tau_b=tau_b,
-            rank=rank,
-            scale_cost=scale_cost,
-            batch_size=batch_size,
-            stage=stage,
-            initializer=initializer,
-            initializer_kwargs=initializer_kwargs,
-            jit=jit,
-            min_iterations=min_iterations,
-            max_iterations=max_iterations,
-            threshold=threshold,
-            linear_solver_kwargs=linear_solver_kwargs,
-            device=device,
+        solve_kwargs = {
+            "epsilon": epsilon,
+            "tau_a": tau_a,
+            "tau_b": tau_b,
+            "rank": rank,
+            "scale_cost": scale_cost,
+            "batch_size": batch_size,
+            "stage": stage,
+            "initializer": initializer,
+            "initializer_kwargs": initializer_kwargs,
+            "jit": jit,
+            "min_iterations": min_iterations,
+            "max_iterations": max_iterations,
+            "threshold": threshold,
+            "device": device,
             **kwargs,
+        }
+
+        # convert problem type to linear for alpha=0
+        if alpha == 0.0 and self.problem_kind != "linear":
+            if self.filtered_vars and len(self.filtered_vars) == 0:
+                raise ValueError("Expected `var_names` to be non-empty for a :attr:`alpha`=0.")
+            logger.info("Ignoring quadratic terms for :attr:`alpha=0`.")
+            self._problem_kind = "linear"
+            for _, value in self.problems.items():
+                value._x = None
+                value._y = None
+                value._problem_kind = "linear"
+
+        # prepare solver kwargs, depending on the problem type
+        if self.problem_kind == "linear":
+            if alpha is not None and alpha > 0:
+                logger.warning("Ignoring :attr:`alpha` for `'linear'` problems.")
+            if "lse_mode" in linear_solver_kwargs:
+                solve_kwargs["lse_mode"] = linear_solver_kwargs["lse_mode"]
+            if "inner_iterations" in linear_solver_kwargs:
+                solve_kwargs["inner_iterations"] = linear_solver_kwargs["inner_iterations"]
+        else:
+            if alpha is None:
+                raise ValueError("Expected :attr:`alpha` to be in interval `[0, 1]`, found `None`.")
+            solve_kwargs["alpha"] = alpha
+            solve_kwargs["linear_solver_kwargs"] = linear_solver_kwargs
+
+        return super().solve(  # type: ignore[return-value]
+            **solve_kwargs,
```

Comment:
I tried and this also works. I'd prefer letting parent classes handle the incompatible arguments with alpha. Some cases of the tests fail since alpha=1.0 and xy is given. It would be also cool to assert that the tests send the desired error message: https://stackoverflow.com/questions/23337471/how-do-i-properly-assert-that-an-exception-gets-raised-in-pytest

```python
        additonal_kwargs = {}
        if alpha is not None and alpha > 0.0:
            additonal_kwargs["alpha"] = alpha
            additonal_kwargs["linear_solver_kwargs"] = linear_solver_kwargs
        else:
            additonal_kwargs.update(linear_solver_kwargs)
        return super().solve(  # type: ignore[return-value]
            epsilon=epsilon,
            tau_a=tau_a,
            tau_b=tau_b,
            rank=rank,
            scale_cost=scale_cost,
            batch_size=batch_size,
            stage=stage,
            initializer=initializer,
            initializer_kwargs=initializer_kwargs,
            jit=jit,
            min_iterations=min_iterations,
            max_iterations=max_iterations,
            threshold=threshold,
            device=device,
            **kwargs,
            **additonal_kwargs,
        )
```

Reply:
Thanks for the feedback @selmanozleyen! There's an important case in which this does not work: when the user specifies both the `sc_attr` and the `spatial_key` in the `prepare` method, but then passes `alpha=0` in `solve`. Here's why that case is important: 

For some downstream analysis functions, the `spatial_key` needs to be set (e.g. `spatial_correspondence`), even though we might not use this data directly for solving the mapping problem. We cannot currently set just the `spatial_key` but not the `sc_attr`, as the backend requires either both or none of them. 

So in that case, we prepare a quadratic problem, we do not modify it or pass `alpha`, and upon calling `solve`, we get `ValueError: Unable to solve a quadratic problem with `alpha = 1` and `xy` supplied.`

If we pass `additonal_kwargs["alpha"] = 0` also in the `else` clause in the `solve` method, we get `ValueError: Expected `alpha` to be in interval `(0, 1]`, found `0`.`. I'm not sure how to go about this. 

I think we need a mechanism to set the `spatial_key`, while still either preparing, or somehow converting, to a linear problem, rather than a quadratic one.

Reply:
One way to go about this would be to handle everything in the `prepare` method, and to make the problem type entirely dependent on `sc_attr`: 
- if it's specified, we're solving a quadratic problem. `alpha=0` is not allowed in `solve`. 
- if it's `None`, we're solving a linear problem. `alpha` is ignored in `solve` (and we can warn about this). 

This way, the `solve` method cannot change the problem type. The problem type is determined on `prepare`, and it depends entirely on `sc_attr`. I pushed a corresponding suggestion, let me know what you think.

Reply:
What you suggested seems reasonable. In the future we might also require a case where we want to specify both but still solve it as a linear problem. I think in that case we would've take the problem type as an argument for prepare. But since sc_attr seems to imply the problem type there is no such need. Either case is better than changing the problem type imo.

Reply:
makes sense!